torch2jax/__init__.py

```diff
@@ -211,6 +218,35 @@ def _coerce(x):
     )
 
 
+def _tree_coerce(x):
+  spec = torch_tree_structure(x)
+  specs = deque([spec])
+
+  while len(specs) > 0:
+    spec = specs.popleft()
+    specs.extend(spec.children_specs)
+    if spec.is_leaf():
+      continue
+    if spec.type in jax._src.tree_util._registry:
+      continue  # already registered
+    node = torch.utils._pytree.SUPPORTED_NODES.get(spec.type, None)
+    if node is None:
+      continue
+    if is_dataclass(spec.type):
+      jax.tree_util.register_dataclass(spec.type)
```

Comment:
is there an advantage to doing this instead of `jax.tree_util.register_pytree_node`?

Reply:
I actually don't know, without specializing it also works for the test cases.